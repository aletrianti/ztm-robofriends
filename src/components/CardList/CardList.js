import React, { Component, Fragment } from 'react';
import Card from '../Card/Card';

class CardList extends Component {
    render() {
        const { robots } = this.props;

        const card = robots.map((robot, i) => {
            return (<Card 
                        key={i}
                        id={robots[i].id}
                        name={robots[i].name}
                        email={robots[i].email}
                    />);
        });
        return (
            <Fragment>
                { card }
            </Fragment>
        );
    }
}

export default CardList;