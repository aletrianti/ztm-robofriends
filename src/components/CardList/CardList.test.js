import React from 'react';
import CardList from './CardList';

it('expects to render CardList component', () => {
    const mockedRobots = [
        {
            id: 1,
            name: 'Robot 1',
            username: 'robot1',
            email: 'robot@one.com'
        }
    ];

    expect(<CardList robots={mockedRobots}/>).toMatchSnapshot();
});