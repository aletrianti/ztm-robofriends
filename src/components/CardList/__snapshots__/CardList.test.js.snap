// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render CardList component 1`] = `
<CardList
  robots={
    Array [
      Object {
        "email": "robot@one.com",
        "id": 1,
        "name": "Robot 1",
        "username": "robot1",
      },
    ]
  }
/>
`;
